source/fundamentals/connection/csot.txt

```diff
@@ -0,0 +1,227 @@
+
+- `commitTransaction()
+  <{+api+}/classes/ClientSession.html#commitTransaction>`__
+- `abortTransaction()
+  <{+api+}/classes/ClientSession.html#abortTransaction>`__
+- `withTransaction() <{+api+}/classes/ClientSession.html#withTransaction>`__
+- `endSession()
+  <{+api+}/classes/ClientSession.html#endSession>`__
+
+If you do not specify ``defaultTimeoutMS``, the driver uses the ``timeoutMS``
+value set on the parent ``MongoClient``.
+
+You cannot override ``defaultTimeoutMS`` by setting the ``timeoutMS`` option on an
+operation in a transaction session provided by the ``withTransaction()`` callback.
+Doing so throws an error.
+
+Client Encryption
+~~~~~~~~~~~~~~~~~
+
+When you use Client-Side Field Level Encryption (CSFLE), the driver uses the
+``timeoutMS`` option to limit the time allowed for encryption and decryption
+operations.
+
+If you specify the ``timeoutMS`` option when you construct a
+``ClientEncryption`` instance, it controls the lifetime of all operations
+performed on that instance. If you do not provide ``timeoutMS``, the instance
+inherits the ``timeoutMS`` setting from the ``MongoClient`` used in the
+``ClientEncryption`` constructor.
+
+If you set ``timeoutMS`` on both the client and directly in
+``ClientEncryption``, the value provided to ``ClientEncryption`` takes
+precedence.
+
+Cursors
+-------
+
+Cursors require special handling when you use the CSOT feature. You can configure
+cursors to interact with CSOT by using either the cursor lifetime or cursor
+iteration mode.
+
+Cursor Lifetime Mode
+~~~~~~~~~~~~~~~~~~~~
+
+The cursor lifetime mode uses ``timeoutMS`` to limit the entire lifetime of a
+cursor. This is the default timeout mode for non-tailable cursors, such as those
+returned by the ``find()``, ``aggregate()``, and ``listCollections()`` methods.
+In this mode, the initialization of the cursor and all subsequent calls to the
+``getMore()`` method must finish within the limit specified by the ``timeoutMS``
+option. If they do not, the system throws a timeout error. 
+
+When you close a cursor by calling the  ``toArray()`` or ``close()`` method, the
+timeout resets.
+
+The following example shows how to set the ``timeoutMS`` option to ensure that
+the cursor is initialized and all documents are retrieved within 10 seconds:
+
+.. code-block:: javascript
+
+   const docs = await collection.find({}, {timeoutMS: 10000}).toArray();
+
+Cursor Iteration Mode
+~~~~~~~~~~~~~~~~~~~~~
+
+The cursor iteration mode uses the ``timeoutMS`` option to limit each call to
+the ``next()``, ``hasNext()``, or ``tryNext()`` method. The timeout refreshes
+after each call completes. This is the default mode for all tailable cursors,
+such as the tailable cursors returned by the ``find()`` method on capped
+collections or change streams.
+
+As shown in the following example, the cursor continues to fetch new documents
+as they are added to a collection and times out if it takes longer than 10
+seconds to retrieve documents:
```

Comment:
```suggestion
as they are added to a collection and times out if it takes longer than 10
seconds to retrieve a document:
```

Practically speaking, the timeout is only going to be "counted down" when fetching a batch of documents, so this is accurate. However, the intention is that from the user's perspective, each next call (so per document) is bounded by the timeoutMS value.